api/controllers/dapps.js

```diff
@@ -43,25 +44,50 @@ DappsController.getDapps = function(context, next) {
 	const params = context.request.swagger.params;
 
 	let filters = {
-		transactionId: params.transactionId.value,
-		name: params.name.value,
+		id: params.transactionId.value,
+		dapp_name: params.name.value,
+	};
+
+	let options = {
 		sort: params.sort.value,
 		limit: params.limit.value,
 		offset: params.offset.value,
+		extended: true,
 	};
 
 	// Remove filters with null values
 	filters = _.pickBy(filters, v => !(v === undefined || v === null));
 
-	modules.dapps.shared.getDapps(_.clone(filters), (err, data) => {
-		try {
-			if (err) {
-				return next(err);
-			}
+	// Remove options with null values
+	options = _.pickBy(options, v => !(v === undefined || v === null));
+
+	// We don't want to change the API so we fix the sort field name here
+	options.sort = options.sort.replace('name', 'dapp_name');
 
+	filters =
+		Object.keys(filters).length > 0
+			? Object.keys(filters).map(aFilter => ({
```

Comment:
Can you simplify?

Reply:
I reckon this should work:

```
const filters = [];

if (params.transactionId.value) {
	filters.push({
		id: params.transactionId.value,
		type: transactionTypes.DAPP,
	});
}

if (params.name.value) {
	filters.push({
		dapp_name: params.name.value,
		type: transactionTypes.DAPP,
	});
}
```

Reply:
@lsilvs yes that will do the same as the current code would you guys prefer it this way? @MaciejBaj

Reply:
I find it easier to understand.

Reply:
Done. Resolving.